fs/fs_zenfs.cc

```diff
@@ -487,17 +487,21 @@ std::shared_ptr<ZoneFile> ZenFS::GetFile(std::string fname) {
 }
 
 /* Must hold files_mtx_ */
-IOStatus ZenFS::DeleteZoneFileNoLock(std::string fname) {
+IOStatus ZenFS::DeleteFileNoLock(std::string fname, const IOOptions& options,
```

Comment:
Have general concerns about the naming convention xyzNoLock().

Just noticed in the code that there are functions which follow exactly the same template (Lock to be held before calling the function) but they are differently named as xyzLocked() - examples are RollMetaZoneLocked(), WriteSnapshotLocked().

Then there are functions which again have similar template but have neither Locked nor NoLock suffix - example GetFileInternal().

Going back to the comment that I made earlier today in the Teams chat - this naming scheme is not very intuitive - we are naming functionality with similar pattern differently at different times. Let's conclude on a common scheme that's explains this best and document it for future use and clarity. @aravind-wdc , @MaisenbacherD , @metaspace , @yhr

Reply:
We should definitely settle on a scheme.

Of the three options, I think that  `xyzNoLock()` is the best one. It communicates that the operation will be performed without taking any locks. I also like `xyzLocked()` but it could be misinterpreted to mean that the operation will be performed locked in the sense that the method will take any locks it needs. `xyzInternal()` is not good, it says nothing about lock invariants.

Reply:
@indraneel-m 
The `xyzLocked()`  naming conventions probably needs some of its own refactoring, which we can take up next.

Reply:
Let's go with the NoLock naming scheme, and follow up with a PR aligning the rest of the code with this

Reply:
Sounds good..